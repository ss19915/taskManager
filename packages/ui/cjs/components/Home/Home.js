"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _theme = require("@react-web-app/theme");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var Home = function Home() {
  return _react["default"].createElement(_theme.Paper, null, "Home");
};

var _default = Home;
exports["default"] = _default;