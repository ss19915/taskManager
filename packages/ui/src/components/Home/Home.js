import React from 'react';
import { Paper } from '@react-web-app/theme';

const Home = () => {

    return (
        <Paper>Home</Paper>
    );
}

export default Home;