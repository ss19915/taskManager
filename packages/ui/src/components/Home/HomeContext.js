import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import _ from 'lodash';
import constants from '../../constants';
import { Box, LinearProgress } from '@task-manager/theme';

const Home = React.lazy(() => import('./Home'));

const HomeContext = (props) => {
    const {
        history,
        user,
        location,
    } = props;
    const { ROUTES: { SIGN_IN } } = constants;

    if (_.isEmpty(user) && location.pathname !== SIGN_IN) {
        history.push(SIGN_IN)
    }
    const Loader = () => (
        <Box mt='40vh'>
            <br />
            <LinearProgress />
            <br />
            <LinearProgress color='secondary' />
            <br />
            <LinearProgress />
            <br />
        </Box>
    );

    return (
        <React.Suspense fallback={<Loader />}>
            <Home />
        </React.Suspense>
    );
}

const mapStateToProps = (state) => ({
    user: state.user
});

export default connect(mapStateToProps)(withRouter(HomeContext));