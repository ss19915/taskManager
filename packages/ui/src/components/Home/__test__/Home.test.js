import React from 'react';
import Home from '../Home';

describe('Component: Home', () => {
    it('Shallow Snapshot test', () => {
        const wrapper = shallow(<Home/>);

        expect(wrapper).toMatchSnapshot();
    });
})