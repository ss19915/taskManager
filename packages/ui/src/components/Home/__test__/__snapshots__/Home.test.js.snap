// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Home Shallow Snapshot test 1`] = `
<WithStyles(ForwardRef(Paper))>
  Home
</WithStyles(ForwardRef(Paper))>
`;
